packages/upswyng-server/src/routes/api/weather.ts

```diff
@@ -0,0 +1,113 @@
+import axios from "axios";
+
+const getCurrentWeather = async (
+  lat: number | null,
+  lon: number | null
+): Promise<TWeatherCurrentResponse> => {
+  try {
+    const { data } = await axios.get<TWeatherCurrentResponse>(
+      "https://api.openweathermap.org/data/2.5/weather?units=imperial",
+      {
+        params: {
+          lat: lat,
+          lon: lon,
+          APPID: process.env.REACT_APP_OPEN_WEATHER_API_KEY,
+        },
+      }
+    );
+    return data;
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+class WeatherCache {
+  millisecondsToExp: number;
+  fetchFunction: typeof getCurrentWeather;
+  cache: TWeatherCurrentResponse;
+  fetchDate: Date;
+  constructor(fetchFunction: () => Promise<TWeatherCurrentResponse>) {
+    this.millisecondsToExp = 600000;
```

Comment:
consider making this an argument w/ a default value (ie `constructor( ....  , msToExpire: number = 60000)`)